sopel/irc/backends.py

```diff
@@ -429,21 +427,24 @@ async def _run_forever(self) -> None:
             # tell the bot to quit without restart
             self.bot.hasquit = True
             self.bot.wantsrestart = False
-            return
         except TimeoutError as err:
             LOGGER.error('Unable to connect due to a timeout: %s', err)
             self.log_exception()
             # tell the bot to quit without restart
             self.bot.hasquit = True
             self.bot.wantsrestart = False
-            return
 
-        # Generic connection error (OSError is used for any connection error)
+        # Generic connection error
+        except ConnectionError as err:
+            LOGGER.error('Unable to connect: %s', err)
+            self.log_exception()
+            # tell the bot to quit without restart
+            self.bot.hasquit = True
+            self.bot.wantsrestart = False
+
+        # Generic OSError (used for any unspecific connection error)
```

Comment:
Nit: I would just leave off the comment here, personally, `OSError` speaks for itself pretty well imo

```suggestion
```

Reply:
I do that sometimes too, but I would probably spell that `# everything else` in this case